src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -159,5 +160,22 @@ public int HResult
         public new Type GetType() => base.GetType();
 
         partial void RestoreRemoteStackTrace(SerializationInfo info, StreamingContext context);
+
+        [StackTraceHidden]
+        internal void SetCurrentStackTrace()
+        {
+            if (_stackTrace != null || _remoteStackTraceString != null)
+            {
+                ThrowHelper.ThrowInvalidOperationException();
+            }
+
+            if (!IsImmutableAgileException(this))
```

Comment:
`IsImmutableAgileException` is CoreCLR specific concept. This part should be in CoreCLR.cs

Reply:
Actually, this whole method should be in CoreCLR.cs

Reply:
Ok.